src/Nix/NixString.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE DeriveGeneric #-}
+module Nix.NixString (
+    NixString
+  , hackyStringHasContext
+  , hackyStringIgnoreContextMaybe
+  , hackyStringIgnoreContext
+  , makeNixStringWithoutContext
+  , modifyNixContents
```

Comment:
I think I'd prefer a `hacky` version of this as well, most of these are probably correct as-is but some string modifications absolutely need to change context as well.